### PR TITLE
Implement direct loads split-K GEMM kernel

### PR DESCRIPTION
**Pull request by bwroblew**
This PR implements a new split-K GEMM kernel that uses LDS direct loads for loading A and B matrices.

Benchmarked it on 30 important split-K GEMM cases using fp16 datatype and MK NK layout (the only one supported by direct loads).
Compared to the best already existing instance, new instances are on average 22% faster with some cases being sped up by 40-95%.

### CODE REVIEW

File: include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_splitk_lds_direct_load.hpp [hunk 1]
```diff
@@ -0,0 +1,999 @@
...
+          bool BThreadTransferSrcResetCoordinateAfterRun,
+          bool BBlockLdsExtraN,
+          index_t CShuffleMRepeatPerShuffle,
+          index_t CShuffleNRepeatPerShuffle,
+          index_t CBlockTransferScalarPerVector_NWaveNPerXDL,
+          typename CBlockTransferClusterLengths_MBlock_MPerBlock_NBlock_NPerBlock,
+          LoopScheduler LoopSched     = make_default_loop_scheduler(),
+          PipelineVersion PipelineVer = PipelineVersion::v1,
+          typename ComputeType        = FloatC>
+struct GridwiseGemm_bk0mk1_bk0nk1_mn_xdlops_splitk_lds_direct_load
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_splitk_lds_direct_load.hpp]:
```suggestion
struct GridwiseGemm_xdlops_splitk_lds_direct_load
```

File: include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_splitk_lds_direct_load.hpp [hunk 2]
```diff
@@ -0,0 +1,999 @@
...
+    }
+
+    // return block_id to C matrix tile idx (m0, n0) mapping
+    template <typename CGridDesc>
+    __host__ __device__ static constexpr auto MakeCBlockClusterAdaptor(
+        const CGridDesc& c_m_n_grid_desc, index_t /* M01 */, index_t /* N01 */, index_t KBatch)
+    {
+        return BlockToCTileMap_KSplit_M00_N0_M01Adapt<MPerBlock, NPerBlock, CGridDesc>(
+            c_m_n_grid_desc, 8, KBatch);
+    }
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_splitk_lds_direct_load.hpp]: This function is probably unused.

File: include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_splitk_c_shuffle_lds_direct_load.hpp [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+          ck::index_t ABlockTransferSrcVectorDim,
+          ck::index_t ABlockTransferSrcScalarPerVector,
+          //   ck::index_t ABlockTransferDstScalarPerVector_K1,
+          bool ABlockLdsAddExtraM,
+          typename BBlockTransferThreadClusterLengths_K0_N_K1,
+          //   typename BBlockTransferThreadClusterArrangeOrder,
+          typename BBlockTransferSrcAccessOrder,
+          ck::index_t BBlockTransferSrcVectorDim,
+          ck::index_t BBlockTransferSrcScalarPerVector,
+          //   ck::index_t BBlockTransferDstScalarPerVector_K1,
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_splitk_c_shuffle_lds_direct_load.hpp]: Are those commented out lines needed ?

File: include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_splitk_c_shuffle_lds_direct_load.hpp [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+        // ABlockTransferDstScalarPerVector_K1,
+        false, // AThreadTransferSrcResetCoordinateAfterRun,
+        ABlockLdsAddExtraM,
+        BBlockTransferThreadClusterLengths_K0_N_K1,
+        // BBlockTransferThreadClusterArrangeOrder,
+        BBlockTransferSrcAccessOrder,
+        BBlockTransferSrcVectorDim,
+        BBlockTransferSrcScalarPerVector,
+        // BBlockTransferDstScalarPerVector_K1,
+        false, // BThreadTransferSrcResetCoordinateAfterRun,
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_splitk_c_shuffle_lds_direct_load.hpp]: As above - are those commented lines needed?

File: include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_splitk_c_shuffle_lds_direct_load.hpp [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+            {LoopScheduler::Default, "Default"}, {LoopScheduler::Interwave, "Interwave"}};
+
+        std::map<PipelineVersion, std::string> PipelineVersionToString{{PipelineVersion::v1, "v1"},
+                                                                       {PipelineVersion::v2, "v2"}};
+
+        str << GridwiseGemm::GetTypeString() << " LoopScheduler: " << LoopSchedToString[LoopSched]
+            << ", PipelineVersion: " << PipelineVersionToString[PipelineVer];
+
+        return str.str();
+    }
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_splitk_c_shuffle_lds_direct_load.hpp]: You should return here the device op name and some of template parameters' values similar as in other device operators.

**Review by aosewski — APPROVED** [review]

**Comment by bartekxk** [comment]
Checked with ninjatracing. Compilation time is under average.

**Comment by bwroblew** [comment]
> Checked with ninjatracing. Compilation time is under average.

Thanks, Bartek!

**Pull request merged**